torchgeo/datasets/nccm.py

```diff
@@ -0,0 +1,244 @@
+    md5s = {"main": "eae952f1b346d7e649d027e8139a76f5"}
+
+    years = [2017, 2018, 2019]
+
+    cmap = {
+        0: (0, 255, 0, 255),
+        1: (255, 0, 0, 255),
+        2: (255, 255, 0, 255),
+        3: (128, 128, 128, 255),
+    }
+
+    def __init__(
+        self,
+        paths: Union[str, Iterable[str]] = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        years: list[int] = [2017, 2018, 2019],
+        classes: list[int] = list(cmap.keys()),
+        transforms: Optional[Callable[[dict[str, Any]], dict[str, Any]]] = None,
+        cache: bool = True,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new dataset.
+
+        Args:
+            paths: one or more root directories to search or files to load
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            years: list of years for which to use dataset
+            classes: list of classes to include, the rest will be mapped to 0
+                (defaults to all classes)
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 after downloading files (may be slow)
+
+        Raises:
+            AssertionError: if ``years`` or ``classes`` are invalid
+            FileNotFoundError: if no files are found in ``paths``
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        assert all(
+            year in self.years for year in years
+        ), f"NCCM data product only exists for the following years: {self.years}"
+
+        assert (
+            set(classes) <= self.cmap.keys()
+        ), f"Only the following classes are valid: {list(self.cmap.keys())}."
+
+        self.paths = paths
+        self.years = years
+        self.classes = classes
+        self.download = download
+        self.checksum = checksum
+        self.ordinal_map = torch.zeros(max(self.cmap.keys()) + 1, dtype=self.dtype)
+        self.ordinal_cmap = torch.zeros((len(self.classes), 4), dtype=torch.uint8)
+
+        curr_path = os.getcwd()
+        curr_path += "/data"
+        if not os.path.exists(curr_path):
+            os.mkdir(curr_path)
+        else:
+            contents = os.listdir(curr_path)
+            for item in contents:
+                item_path = os.path.join(curr_path, item)
+                if os.path.isfile(item_path):
+                    os.remove(item_path)
+                elif os.path.isdir(item_path):
+                    shutil.rmtree(item_path)
```

Comment:
What is this and why is it needed?

Reply:
the code we had written earlier wasn't creating the data directory to store the files by default for some reason. So here we are just creating the directory if it doesn't exist, and if it does exist, we are just clearing out the old content.

Reply:
> for some reason

We should figure out the reason and fix that instead.